gologger/LogManager.go

```diff
@@ -155,81 +157,83 @@ func (l *CustomLogger) GetLogLevel() LogLevels {
 	return l.logLevel
 }
 
-// LogErrorInterface is used to send errors
+// LogErrorInterface is used to log errors
 func (l *CustomLogger) LogErrorInterface(v ...interface{}) {
 	l.logger.Output(2, fmt.Sprint(v...))
 }
 
-// LogError is used to send errors and a message along with the error
+// LogError is used to log errors and a message along with the error
 func (l *CustomLogger) LogError(str string, err error) {
-	l.logger.Printf(`{"log_level": %q, "log_timestamp": %q, "log_facility": %q,"log_message": %q,"log_error": %q,"K8sNamespace": %q}`,
-		ERROR.String(), time.Now().String(), l.graylogFacility, str, err.Error(), l.k8sNamespace)
+	pairs := []Pair{
+		{"log_error", err.Error()},
+	}
+	l.logMessageWithExtras(str, ERROR, pairs)
 }
 
-// LogErrorWithoutError is used to send only a message and not an error
+// LogErrorWithoutError is used to log only a message and not an error
 func (l *CustomLogger) LogErrorWithoutError(str string) {
-	l.logMessage(str, ERROR)
+	l.logMessageWithExtras(str, ERROR, nil)
 }
 
-// LogErrorWithoutErrorf is used to send only a message and not an error
+// LogErrorWithoutErrorf is used to log only a message and not an error
 func (l *CustomLogger) LogErrorWithoutErrorf(str string, args ...interface{}) {
 	l.LogErrorWithoutError(fmt.Sprintf(str, args...))
 }
 
-//LogErrorMessage is used to send extra fields to graylog along with the error
+// LogErrorMessage is used to log extra fields to graylog along with the error
 func (l *CustomLogger) LogErrorMessage(str string, err error, pairs ...Pair) {
 	if err != nil {
 		pairs = append(pairs, Pair{"log_error", err.Error()})
 	}
 	l.logMessageWithExtras(str, ERROR, pairs)
 }
 
-// LogWarning is used to send warning messages
+// LogWarning is used to log warning messages
 func (l *CustomLogger) LogWarning(str string) {
 	if l.logLevel >= WARN {
-		l.logMessage(str, WARN)
+		l.logMessageWithExtras(str, WARN, nil)
```

Comment:
what does nil stand for? if it is supposed to be nil, why change the method?

Reply:
`logMessageWithExtras` function expects list of KV pairs which can be logged, I'm setting nil here saying I don't have any pairs to send just log the warn message. The `logMessage` function is very much similar to `logMessageWithExtras` function the only difference is `logMessage` function only formats the log message and logs it while we already have a function that formats as well as include custom KV pairs as argument. So thought of removing `logMessage` function and use one generic function `logMessageWithExtras` everywhere.